nntrainer/src/tensor.cpp

```diff
@@ -775,4 +811,58 @@ Tensor Tensor::standardization() const {
 
   return result;
 }
+
+ExternalLoopInfo Tensor::computeExternalLoop(const Tensor &m) {
+  if (m.length() > this->length())
+    throw exception::not_supported("broadcasting *this is not supported");
+
+  const TensorDim m_dim = m.getDim();
+
+  ExternalLoopInfo e;
+
+  for (unsigned int i = 0; i < MAXDIM; ++i) {
+    if (m_dim.getTensorDim(i) == 1) {
+      e.strides[i] = 0;
+      continue;
+    }
+
+    if (dim.getTensorDim(i) == m_dim.getTensorDim(i)) {
+      e.strides[i] = m.strides[i];
+      continue;
+    }
+
+    std::stringstream ss;
+    ss << "[computeExternalLoop] given shape is not broadcastable,\n"
```

Comment:
Update to `broadcasting only allowed for dimension value of only 1`. This is gives the reason why this wont work.

Reply:
Done